### PR TITLE
Flatten json

### PR DESCRIPTION
**Pull request by manycoding**
This one significantly increases performance and decreases memory usage if using `flat_df`, which corresponds to `expand=True` default parameter.
Some things to consider:
1. If original column name and expanded column name the same, the original column will be dropped. 
See here https://github.com/amirziai/flatten/issues/48
I think it won't happen in 99% of cases, I've never encountered such a case. Alternatively, we can set other separator which potentially bumps it to 99.9%.
Nevertheless, there's is that issue and also a comment in tests.

Time and peak memory, 1k samples from a heavily nested df (see `shared/qa/Experiments/data/items_epicsports_150k_nested.pickle`).
This pr - 1.63s, 2400 MB
Old - 4min 57s, 4411.51 MiB
#74 - 6min 2s, 2800 MB

And, interestingly, all 150k items with this pr:
9min 40s
peak memory: 3682.77 MiB, increment: 1391.34 MiB

### CODE REVIEW

File: src/arche/readers/items.py [hunk 1]
```diff
@@ -13,23 +14,25 @@
...
         return len(self.df)
 
     @property
-    def flat_df(self):
+    def flat_df(self) -> pd.DataFrame:
         if self._flat_df is None:
             if self.expand:
-                self._flat_df, self._columns_map = pandas.flatten_df(self.df)
+                self._flat_df = pd.DataFrame(flatten(i) for i in self.raw)
+                self._flat_df["_key"] = self.df.get(
```
Comment (manycoding) [inline src/arche/readers/items.py]: We need the same formatted keys as in original df.

File: src/arche/readers/items.py [hunk 2]
```diff
@@ -40,8 +43,12 @@ def process_df(df: pd.DataFrame) -> pd.DataFrame:
             df["_type"] = df["_type"].astype("category")
         return df
 
-    def get_origin_column_name(self, column_name: str) -> str:
-        return self._columns_map.get(column_name, column_name)
+    def origin_column_name(self, new: str) -> str:
```
Comment (manycoding) [inline src/arche/readers/items.py]: KISS approach. I believe it will work in most cases.

File: tests/readers/test_items.py [hunk 3]
```diff
@@ -134,3 +133,78 @@ def test_process_df():
...
+        {"links_0_Instagram_0": "links", "links_1_ITW website_0": "links"},
+    ),
+    # Corner case https://github.com/amirziai/flatten/issues/48
+    (
+        [
+            {"type": [0], "str": "k", "type_0": 5},
+            {"type": [0, [2, 3]], "str": "s", "type_0": 6},
+        ],
+        {
+            "str": ["k", "s"],
```
Comment (manycoding) [inline tests/readers/test_items.py]: Here we lost `type 0,0`
